checks/system/integration/utils.py

```diff
@@ -0,0 +1,213 @@
+        self.CLASS_INCLUDE = []
+        # Print what check will be created. Do nothing.
+        self.DEBUG         = False
+        # The computing system we expect to run on.
+        self.SYSTEM        = None
+        # The name of the calling module. Should be set from the caller.
+        self.MODULE_NAME   = __name__
+
+    def __call__(self, cmd, expected=None, not_expected=None, where=''):
+        """
+        Create a test of 'cmd'. A regex describing the expected output
+
+        :param cmd:             The command and its arguments to execute. This
+                                may include shell pipes and redirections.
+        :param expected:        A string or [string,string]. In the first case,
+                                string is a regular expression describing the
+                                expected output.  In the second case, the
+                                first string is the regex and the second string
+                                is either 'stdout' or 'stderr', depending on
+                                where the regex should search. The default is
+                                'stdout'.  If None, or missing, no search will
+                                be performed.
+        :param not_expected:    Similar to 'expected', but what should not be
+                                found.
+        """
+
+        def debuginfo():
+            """
+            To help the user, grab the line where this check was created
+            from so we can include it in any error message.
+            """
+            from inspect import getframeinfo, stack
+            caller = getframeinfo(stack()[2][0])
+            return "[%s:%d]" % (caller.filename, caller.lineno)
+
+        #
+        # We intentionally increment here before the inclusion/exclusion tests
+        # so that numbering stays consistent when all tests are enabled.
+        #
+        Check.check_id += 1
+
+        lcl_CHECK_CLASS = ''
+        if self.CLASS:
+            lcl_CHECK_CLASS = f'_{self.CLASS}'
+
+        #
+        # Check if a class of tests has been explicitly included or excluded
+        #
+        exclude = self.CLASS_EXCLUDE
+        include = self.CLASS_INCLUDE
+
+        if exclude or include:
+            if not exclude:
+                exclude = ['*']
+            if not include:
+                include = ['*']
+
+        if '*' not in exclude:
+            if self.CLASS in exclude:
+                return
+        elif '*' not in include:
+            if self.CLASS not in include:
+                return
+
+        #
+        # If where is unspecificed we can run with any feature.
+        # Be forgiving if the user forgets a leading +.
+        #
+
+        if not where:
+            where = '*'
+        elif where[0] not in ['-', '+']:
+            where = f'+{where}'
+
+        #
+        # Get out properties ready.
+        #
+
+        name                = f'Check_{Check.check_id:04}{lcl_CHECK_CLASS}'
```

Comment:
I think the naming is not ideal. As soon as another test is added or deleted then everything will be shifted. It can be hard to track long term data and compare between different systems.
Should the users filter by the name if they are testing one service? Maybe it would make sense to add some tag in the test.

Reply:
Hm the tag will not be appended to the name. I had something like this in mind: https://reframe-hpc.readthedocs.io/en/stable/tutorial.html#test-filtering 

> Another useful filtering option is the [-t](https://reframe-hpc.readthedocs.io/en/stable/manpage.html#cmdoption-t) option, which selects tests by tag. You can assign multiple tags to a test by settings its [tags](https://reframe-hpc.readthedocs.io/en/stable/regression_test_api.html#reframe.core.pipeline.RegressionTest.tags) attribute. You can use tags to effectively categorize tests.

I was thinking something like `slurm`, `fs` etc. You can easily set the check class as a tag (like the name), so you can filter them with `-t`, instead of a regex and `-n`. Maybe not a big improvement, unless you want to make other subgroups.